scripts/build.js

```diff
@@ -84,6 +90,40 @@ function build(options) {
 }
 
 /**
+ * Resolve node-gyp command to invoke
+ *
+ * @api private
+ */
+function resolveNodeGyp() {
+  if (process.jsEngine === 'chakracore') {
+    // For node-chakracore, check if node-gyp is in the path.
+    // If yes, use it instead of using node-gyp directly from
+    // node_modules because the one in node_modules is not
+    // compatible with node-chakracore.
+    var nodePath = path.dirname(process.execPath);
+    var nodeGypName = process.platform === 'win32' ? 'node-gyp.cmd' : 'node-gyp';
+    var globalNodeGypBin = process.env.Path.split(';').filter(function(envPath) {
```

Comment:
> is this correct for non-Windows?

good catch @saper . I have updated it with appropriate delim.

> what is the difference between chakraized node-gyp and the normal one?

See https://github.com/nodejs/node-gyp/pull/873 for more details and the [description](https://github.com/sass/node-sass/pull/1777/commits/eec8afa3c240f26261fc548698b2fe8ae77a131d) of my first commit.

In short, the changes are to add engine specific libs and headers while compiling native modules.

Since `node-gyp` PR has landed, we might not need this. The reason I still want to have this around is we could have a case of having a stale non-chakracore supported `node-gyp` in module path which will break building `node-sass` with `node-chakracore` as it breaks today.